### ISSUE TITLE
[Joss review] Suggestions and request regarding the paper

### ISSUE THREAD

**Issue by Saran-nns** [carnotresearch/cr-sparse#21]
Hi @shailesh1729,

I have some suggestions about the paper. Let me know if you disagree with any of them:

#### Summary:

- [x] The paragraphs between lines 14-61 can be under a separate heading, e.g.: 'Package Overview'. 
- [x] Brief introduction to problems of sparse representation / compressive sensing methods and the need for sparse recovery algorithms and their application domains _for a broad and non-specialist audience_.  If you think it is out of the scope of this paper, refer the reader to the references where they can read the explanations.

#### Statement of need: 

- [x] This section could be used to give a general idea of what to expect from the package in terms of the concepts/problems mentioned in the `Summary` section, which I think are convincingly defined between lines 63 - 76. However, lines 77-83 are subjective. For example. 
   - Line 77: There are some libraries (please cite them) 
   - Line 81: There are several attempts (please cite them) 
   - Line 83: Implementation appears to be somewhat complex (please explain).

- [x] The rest of this section deals with JAX, which I think is beyond the scope of this project.
 
- [x] The package has a broader scope, I suggest adding or naming some target groups. 

### Documentation/Readme

- [x] Please give a brief introduction to sparse signal processing and the need for algorithms and the cr-sparse package. You can just copy this from the 'summary' section of the paper.

- [x] Readme lines 17-40 
For ease of use, it would be very useful to classify these methods either by algorithms (sparse recovery/solvers) or by their applications (compressive sensing/sparse signal processing). Some examples of sparse recovery algorithm classification can be found [here](https://arxiv.org/pdf/1801.09744.pdf#:~:text=2%2C%20sparse%20recovery%20algorithms%20can,problem%20through%20convex%20relaxation%20algorithms.) and [here](https://ieeexplore.ieee.org/document/8577023). IIf you feel that it is not feasible to provide the overview of the algorithms/methods classification in the readme file, I still recommend including it in the documentation and attaching the link in the readme file. 

### General 

- [x] Please provide some limitations of the package and future work. This would let the users know what you currently working on and what they cannot expect from the current version of the package.

**Comment by shailesh1729** [comment]
Dear @Saran-nns,

Thank you very much for the detailed feedback. I have updated the paper draft in openjournals/joss-reviews#3917.  I am also trying to balance between the paper length concerns mentioned by pdebuyl and adding more content to address the concerns listed here. 

Some specific answers:

- I have added a very short intro to sparse problems. I have kept the more detailed mathematical intro in the [online docs](https://cr-sparse.readthedocs.io/en/latest/intro.html).  Also have provided references to relevant books and papers. 
- Have removed JAX description from the statement of need section. Moved some of the discussion into the new limitations section as relevant. 
- See `here <https://cr-sparse.readthedocs.io/en/latest/algorithms.html>`_ in the docs for the list of algorithms supported (and planned) in organized hierarchically. I have added a reference to this page in README.  I will slowly enrich this page further with more information later as I build more detailed usage and application docs for individual algorithms.
